src/ripple/app/tx/impl/XChainBridge.cpp

```diff
@@ -0,0 +1,1675 @@
+        &atts[0], &atts[0] + atts.size(), quorum, signersList);
+
+    if (!createCID)
+    {
+        // Modify the object before it's potentially deleted, so the meta data
+        // will include the new attestations
+        if (!sleCID)
+            return tecINTERNAL;  // LCOV_EXCL_LINE
+        sleCID->setFieldArray(
+            sfXChainCreateAccountAttestations, curAtts.toSTArray());
+        psb.update(sleCID);
+    }
+
+    // Account create transactions must happen in order
+    if (rewardAccounts && claimCount + 1 == attBegin->createCount)
+    {
+        auto const r = finalizeClaimHelper(
+            psb,
+            bridgeSpec,
+            attBegin->toCreate,
+            /*dstTag*/ std::nullopt,
+            doorAccount,
+            attBegin->sendingAmount,
+            /*rewardPoolSrc*/ doorAccount,
+            attBegin->rewardAmount,
+            *rewardAccounts,
+            srcChain,
+            sleCID,
+            OnTransferFail::removeClaim,
+            ctx_.journal);
+        if (!isTesSuccess(r))
+        {
+            if (r == tecINTERNAL || r == tecINSUFFICIENT_FUNDS ||
+                isTefFailure(r))
+                return r;
+        }
+        // Move past this claim id even if it fails, so it doesn't block
+        // subsequent claim ids
+        (*sleB)[sfXChainAccountClaimCount] = attBegin->createCount;
+        psb.update(sleB);
+    }
+    else if (createCID)
+    {
+        if (sleCID)
+            return tecINTERNAL;  // LCOV_EXCL_LINE
+
+        auto const sleCID = std::make_shared<SLE>(claimKeylet);
+        (*sleCID)[sfAccount] = doorAccount;
+        (*sleCID)[sfXChainBridge] = bridgeSpec;
+        (*sleCID)[sfXChainAccountCreateCount] = attBegin->createCount;
+        sleCID->setFieldArray(
+            sfXChainCreateAccountAttestations, curAtts.toSTArray());
+
+        // Add to owner directory of the door account
+        auto const page = ctx_.view().dirInsert(
+            keylet::ownerDir(doorAccount),
+            claimKeylet,
+            describeOwnerDir(doorAccount));
+        if (!page)
+            return tecDIR_FULL;
+        (*sleCID)[sfOwnerNode] = *page;
+
+        // Reserve was already checked
+        adjustOwnerCount(psb, sleDoor, 1, ctx_.journal);
+        psb.insert(sleCID);
+        psb.update(sleDoor);
+    }
+
+    psb.apply(ctx_.rawView());
+
+    return tesSUCCESS;
+}
+
+TER
+XChainAddAttestation::doApply()
+{
+    STXChainAttestationBatch const batch = ctx_.tx[sfXChainAttestationBatch];
+
+    std::vector<TER> applyRestuls;
+    applyRestuls.reserve(batch.numAttestations());
+
+    auto const& bridgeSpec = batch.bridge();
+
+    auto const bridgeK = keylet::bridge(bridgeSpec);
+    // Note: sle's lifetimes should not overlap calls to applyCreateAccountAtt
+    // and applyClaims because those functions create a sandbox `sleB` is reset
+    // before those calls and should not be used after those calls are made.
+    // (it is not `const` because it is reset)
```

Comment:
I've tried to fix this. I agree it would be nicer if I could make it overlap, but everything I can think of makes is worse. Want to write a proposal to make using a moved from variable a compiler error :-) That would solve this.

To be clear, I don't disagree with your comment, I just think we're stuck with this unless we can come up with something cleaner.

Reply:
@scottschurr I added a patch to address this: `7dcebc6f64 [fold] When adding an attestation, if funds don't move still add attestation (and return success). Also limit sle scopes:`

I made a lambda that limits the scope of the sles, and returns a "whole bunch of stuff" from the lambda. Yes, I know it's ugly (and I left comments saying so), but it's the best I could think of.